doc/howto/cluster_config_networks.md

```diff
@@ -0,0 +1,103 @@
+(cluster-config-networks)=
+# How to configure networks for a cluster
+
+All members of a cluster must have identical networks defined.
+The only configuration keys that may differ between networks on different members are [`bridge.external_interfaces`](network-bridge-options), [`parent`](network-external), [`bgp.ipv4.nexthop`](network-bridge-options) and [`bgp.ipv6.nexthop`](network-bridge-options).
+See {ref}`clustering-member-config` for more information.
+
+Creating additional networks is a two-step process:
+
+1. Define and configure the new network across all cluster members.
+   For example, for a cluster that has three members:
+
+       lxc network create --target node1 my-network
+       lxc network create --target node2 my-network
+       lxc network create --target node3 my-network
+
+   ```{note}
+   You can pass only the member-specific configuration keys `bridge.external_interfaces`, `parent`, `bgp.ipv4.nexthop` and `bgp.ipv6.nexthop`.
+   Passing other configuration keys results in an error.
+   ```
+
+   These commands define the network, but they don't create it.
+   If you run `lxc network list`, you can see that the network is marked as "pending".
+1. Run the following command to instantiate the network on all cluster members:
+
+       lxc network create my-network
+
+   ```{note}
+   You can add configuration keys that are not member-specific to this command.
+   ```
+
+   If you missed a cluster member when defining the network, or if a cluster member is down, you get an error.
+
+Also see {ref}`network-create-cluster`.
+
+## Separate REST API and clustering networks
+
+You can configure different networks for the REST API endpoint of your clients and for internal traffic between the members of your cluster.
+This separation can be useful, for example, to use a virtual address for your REST API, with DNS round robin.
+
+To do so, you must specify different addresses for [`core.https_address`](server) (the address for the REST API) and [`cluster.https_address`](server) (the address for internal cluster traffic).
+
+1. Configure the bootstrap server:
```

Comment:
I couldn't figure out how to test this - so please check if the steps make sense.

Reply:
I think the core.https_address and cluster.https_address are the wrong way around, ill test this and let you know.

Reply:
So this isn't quite right. The `cluster.https_address` cannot be changed once a member is in the cluster.

```
lxc shell v1
root@v1:~# lxd init
Would you like to use LXD clustering? (yes/no) [default=no]: yes
What IP address or DNS name should be used to reach this node? [default=10.21.203.10]: 
Are you joining an existing cluster? (yes/no) [default=no]: 
What name should be used to identify this node in the cluster? [default=v1]: 
Setup password authentication on the cluster? (yes/no) [default=no]: 
Do you want to configure a new local storage pool? (yes/no) [default=yes]: 
Name of the storage backend to use (dir, lvm, zfs, btrfs) [default=zfs]: dir
Do you want to configure a new remote storage pool? (yes/no) [default=no]: 
Would you like to connect to a MAAS server? (yes/no) [default=no]: 
Would you like to configure LXD to use an existing bridge or host interface? (yes/no) [default=no]: 
Would you like to create a new Fan overlay network? (yes/no) [default=yes]: 
What subnet should be used as the Fan underlay? [default=auto]: 
Would you like stale cached images to be updated automatically? (yes/no) [default=yes]: 
Would you like a YAML "lxd init" preseed to be printed? (yes/no) [default=no]: 

root@v1:~# lxc config show 
config:
  cluster.https_address: 10.21.203.10:8443
  core.https_address: 10.21.203.10:8443

root@v1:~# lxc cluster add v2
Member v2 join token:
eyJzZXJ2ZXJfbmFtZSI6InYyIiwiZmluZ2VycHJpbnQiOiIzY2NmNDMxZmY1MGVhYzVmY2JjZDgxYjRkZjU2MDVkMGI4ZWViMDg3NWQ3NDU2ZDg5YzZlZmY0ZWY2MzM1ZmNiIiwiYWRkcmVzc2VzIjpbIjEwLjIxLjIwMy4xMDo4NDQzIl0sInNlY3JldCI6ImE2OGU3ZTc4MGUwYTU2YWZkZjFiZDc0NjI2ZTZiMjk3NjJhNThhZWY3ODE2YzFmYTgwNzY3YmVkMTgwZTUzMTkifQ==

root@v1:~# lxc cluster add v3
Member v3 join token:
eyJzZXJ2ZXJfbmFtZSI6InYzIiwiZmluZ2VycHJpbnQiOiIzY2NmNDMxZmY1MGVhYzVmY2JjZDgxYjRkZjU2MDVkMGI4ZWViMDg3NWQ3NDU2ZDg5YzZlZmY0ZWY2MzM1ZmNiIiwiYWRkcmVzc2VzIjpbIjEwLjIxLjIwMy4xMDo4NDQzIl0sInNlY3JldCI6ImU0ODI1ZDYxMTY2OGEzNDI4OWNkNDA4YmIwNTkyZDM0MDFkMTVhZWU4ZDgzMzE1ODMyNGViNWE5YzZmMWEyZTUifQ==
root@v1:~# exit
logout

lxc shell v2
root@v2:~# lxd init
Would you like to use LXD clustering? (yes/no) [default=no]: yes
What IP address or DNS name should be used to reach this node? [default=10.21.203.4]: 
Are you joining an existing cluster? (yes/no) [default=no]: yes
Do you have a join token? (yes/no/[token]) [default=no]: yes
Please provide join token: eyJzZXJ2ZXJfbmFtZSI6InYyIiwiZmluZ2VycHJpbnQiOiIzY2NmNDMxZmY1MGVhYzVmY2JjZDgxYjRkZjU2MDVkMGI4ZWViMDg3NWQ3NDU2ZDg5YzZlZmY0ZWY2MzM1ZmNiIiwiYWRkcmVzc2VzIjpbIjEwLjIxLjIwMy4xMDo4NDQzIl0sInNlY3JldCI6ImE2OGU3ZTc4MGUwYTU2YWZkZjFiZDc0NjI2ZTZiMjk3NjJhNThhZWY3ODE2YzFmYTgwNzY3YmVkMTgwZTUzMTkifQ==
All existing data is lost when joining a cluster, continue? (yes/no) [default=no] yes
Choose "source" property for storage pool "local": 
Would you like a YAML "lxd init" preseed to be printed? (yes/no) [default=no]: 

root@v2:~# lxc config show
To start your first container, try: lxc launch ubuntu:22.04
Or for a virtual machine: lxc launch ubuntu:22.04 --vm

config:
  cluster.https_address: 10.21.203.4:8443
  core.https_address: 10.21.203.4:8443
root@v2:~# exit
logout

lxc shell v3
root@v3:~# lxd init
Would you like to use LXD clustering? (yes/no) [default=no]: yes
What IP address or DNS name should be used to reach this node? [default=10.21.203.5]: 
Are you joining an existing cluster? (yes/no) [default=no]: yes
Do you have a join token? (yes/no/[token]) [default=no]: eyJzZXJ2ZXJfbmFtZSI6InYzIiwiZmluZ2VycHJpbnQiOiIzY2NmNDMxZmY1MGVhYzVmY2JjZDgxYjRkZjU2MDVkMGI4ZWViMDg3NWQ3NDU2ZDg5YzZlZmY0ZWY2MzM1ZmNiIiwiYWRkcmVzc2VzIjpbIjEwLjIxLjIwMy4xMDo4NDQzIl0sInNlY3JldCI6ImU0ODI1ZDYxMTY2OGEzNDI4OWNkNDA4YmIwNTkyZDM0MDFkMTVhZWU4ZDgzMzE1ODMyNGViNWE5YzZmMWEyZTUifQ==
All existing data is lost when joining a cluster, continue? (yes/no) [default=no] yes
Choose "source" property for storage pool "local": 
Would you like a YAML "lxd init" preseed to be printed? (yes/no) [default=no]: 

root@v3:~# lxc config show
To start your first container, try: lxc launch ubuntu:22.04
Or for a virtual machine: lxc launch ubuntu:22.04 --vm

config:
  cluster.https_address: 10.21.203.5:8443
  core.https_address: 10.21.203.5:8443

root@v3:~# lxc config set cluster.https_address=10.21.203.5:8444
Error: Changing cluster.https_address is currently not supported

root@v3:~# lxc config set core.https_address=10.21.203.5:8444

root@v3:~# lxc config show
config:
  cluster.https_address: 10.21.203.5:8443
  core.https_address: 10.21.203.5:8444
```

So its the `cluster.https_address` which must uniquely identify the cluster member and cannot be changed after joining.
Whereas the `core.https_address` is the address that the member's REST API will listen on and can be different and changed on each cluster member. 

Unlike `cluster.https_address` (which must be a specific reachable IP address, as its used by the other members to contact the local member) the `core.https_address` can be a wildcard address so that it listens on multiple interfaces if needed.

Reply:
OK, so it's basically just "Create your cluster as usual, then set core.https_address on some members to a different address."?

Reply:
Yep spot on :)